examples/platform/efr32/TemperatureSensor.h

```diff
@@ -0,0 +1,38 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "sl_status.h"
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+class TemperatureSensor
```

Comment:
This seems to be defined extern "C", so class feels odd. Should we have "extern C" at all?

Since this class only contains static members, should it be a namespace or just smurf-named (like `TemperatureSensor_Init()` and such?

Reply:
The extern C was just needed for for the C implementation for the physical sensor from our gsdk code. I moved that to the cpp file and just for the include.